src/main/java/org/folio/rest/helper/TransactionSummariesHelper.java

```diff
@@ -19,14 +24,25 @@ public TransactionSummariesHelper(Map<String, String> okapiHeaders, Context ctx,
   }
 
   public CompletableFuture<OrderTransactionSummary> createOrderTransactionSummary(OrderTransactionSummary orderSummary) {
-    return VertxCompletableFuture.runAsync(ctx, () -> validateTransactionCount(orderSummary))
+    return VertxCompletableFuture.runAsync(ctx, () -> validateTransactionCount(Arrays.asList(orderSummary.getNumTransactions())))
       .thenCompose(ok -> handleCreateRequest(resourcesPath(ORDER_TRANSACTION_SUMMARIES), orderSummary))
       .thenApply(orderSummary::withId);
   }
 
-  private void validateTransactionCount(OrderTransactionSummary orderSummary) {
-    if (orderSummary.getNumTransactions() < 1) {
-      throw new CompletionException(new HttpException(422, ErrorCodes.INVALID_TRANSACTION_COUNT));
+  public CompletableFuture<InvoiceTransactionSummary> createInvoiceTransactionSummary(InvoiceTransactionSummary invoiceSummary) {
+    return VertxCompletableFuture
+      .runAsync(ctx,
+          () -> validateTransactionCount(
+              Arrays.asList(invoiceSummary.getNumPaymentsCredits(), invoiceSummary.getNumEncumbrances())))
+      .thenCompose(ok -> handleCreateRequest(resourcesPath(INVOICE_TRANSACTION_SUMMARIES), invoiceSummary))
+      .thenApply(invoiceSummary::withId);
+  }
+
+  private void validateTransactionCount(List<Integer> summaryDetailCounts) {
```

Comment:
The method will become simpler and easier to use if it has one argument of **integer** type.

Reply:
Currently, there only 2 methods using it. invoiceSummary has 2 counts. Are you suggesting something like `validateTransactionCount(Integer summaryCounts1, Integer summaryCounts2)` and call it like `validateTransactionCount(orderSummary.getNumTransactions(), null)` ?

Reply:
I'm suggesting something like `validateTransactionCount(Integer summaryCounts)`. And then call it  like:
`validateTransactionCount(invoiceSummary.getNumPaymentsCredits())`, `validateTransactionCount(invoiceSummary.getNumEncumbrances())` or `validateTransactionCount(orderSummary.getNumTransactions())`